packages/client-sdk-nodejs/src/internal/auth-client.ts

```diff
@@ -0,0 +1,136 @@
+import {auth} from '@gomomento/generated-types';
+import grpcAuth = auth.auth;
+import {Header, HeaderInterceptorProvider} from './grpc/headers-interceptor';
+import {ClientTimeoutInterceptor} from './grpc/client-timeout-interceptor';
+import {ChannelCredentials, Interceptor} from '@grpc/grpc-js';
+import {
+  GenerateApiToken,
+  RefreshApiToken,
+  CredentialProvider,
+  MomentoLogger,
+} from '..';
+import {version} from '../../package.json';
+import {IdleGrpcClientWrapper} from './grpc/idle-grpc-client-wrapper';
+import {GrpcClientWrapper} from './grpc/grpc-client-wrapper';
+import {Configuration} from '../config/configuration';
+import {cacheServiceErrorMapper} from '../errors/cache-service-error-mapper';
+import {normalizeSdkError} from '@gomomento/sdk-core/dist/src/errors';
+import {validateValidForSeconds} from '@gomomento/sdk-core/dist/src/internal/utils';
+import Never = grpcAuth._GenerateApiTokenRequest.Never;
+import Expires = grpcAuth._GenerateApiTokenRequest.Expires;
+
+export interface AuthClientProps {
+  configuration: Configuration;
+  credentialProvider: CredentialProvider;
+}
+
+export class AuthClient {
+  private readonly clientWrapper: GrpcClientWrapper<grpcAuth.AuthClient>;
+  private readonly interceptors: Interceptor[];
+  private static readonly REQUEST_TIMEOUT_MS: number = 60 * 1000;
+  private readonly logger: MomentoLogger;
+
+  /**
+   * @param {AuthClientProps} props
+   */
+  constructor(props: AuthClientProps) {
+    this.logger = props.configuration.getLoggerFactory().getLogger(this);
+    const headers = [
+      new Header('Authorization', props.credentialProvider.getAuthToken()),
```

Comment:
This shouldn't be necessary for the Auth APIs, right?

Reply:
cc: @rlinehan @bruuuuuuuce

Reply:
yep, shouldn't be necessary

Reply:
No it shouldn't, just a copy-paste from another client - removing!